common/quotas/global/collection/collection.go

```diff
@@ -163,89 +333,146 @@ func (c *Collection) backgroundUpdateLoop() {
 		case <-ticker.Chan():
 			now := c.timesource.Now()
 			c.logger.Debug("update tick")
-			// update  if it changed
+			// update interval if it changed
 			newTickInterval := c.updateInterval()
 			if tickInterval != newTickInterval {
 				tickInterval = newTickInterval
 				ticker.Reset(newTickInterval)
 			}
 
-			capacity := c.limits.Len()
+			// submit local metrics asynchronously, because there's no need to do it synchronously
+			localMetricsDone := make(chan struct{})
+			go func() {
+				defer close(localMetricsDone)
+				c.local.Range(func(k string, v internal.CountedLimiter) bool {
+					globalKey := c.km.LocalToGlobal(k)
+					counts := v.Collect()
+					if counts.Idle > gcAfterIdle || c.shouldDeleteKey(c.keyMode(globalKey), true) {
+						c.logger.Debug(
+							"deleting local ratelimiter",
+							tag.Dynamic("key", globalKey),
+							tag.Dynamic("idle", counts.Idle > gcAfterIdle),
+						)
```

Comment:
I liked having these for local monitoring, but they feel remove-able for merging since this is a relatively hot loop (though not performance-sensitive).

I'm not sure how we all feel about debug-oriented logs tbh.  Any votes?